xdcr.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2024-Present Couchbase, Inc.
```

Comment:
I think it should be made more obvious that this code isn't XDCR itself, it's an abstraction of XDCR management to support testing.  I'd probably name it something like xdcr_manager.go as part of that, rename 
"XDCR" to XDCRManager, and have a high-level description of how it's intended to be used at the top of the file, including something about the fact that it's intended to support Couchbase and rosmar implementations.